pyk/src/pyk/proof/proof.py

```diff
@@ -303,20 +304,110 @@ def lines(self) -> list[str]:
         return [line for lines in (summary.lines for summary in self.summaries) for line in lines]
 
 
-class ProofStep: ...
+class FailureInfo: ...
 
 
-class StepResult: ...
+class ProverPool(ContextManager['ProverPool'], Generic[P, PS, SR]):
```

Comment:
What are the generic parameters here? I'm not super familiar with this code so it may be more obvious with context, but it would be nice to document things like this in Sphinx as well!